Project01/BankAccount.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lesson06
+{
+    public enum BType { personal, company };
+    internal class BankAccount
+    {
+        private static int _number;
+        public int Number { get; }
+
+        private BType _type;
+
+        private decimal _sum;
+        public decimal Sum { get { return _sum; } set { _sum = value; } }
+
+        internal BankAccount(BType type)
+        {
+            _number++;
+            Number = _number;
+            this._type = type;
+        }
+
+        public static bool operator ==(BankAccount a1, BankAccount a2)
+        {
+            return a1._sum == a2._sum && a1._type == a2._type;
```

Comment:
А если в `a1`, или в `a2` будет `null`? Всё сломается?
```suggestion
            return Equals(a1, a2);
```